expo/eas-cli
# Prompt to add bundle identifier and application id

# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary.

# Why

- Resolve https://linear.app/expo/issue/ENG-894/automatically-add-bundleidentifier-and-package-when-running-eas-build

# How

- Copy over code from expo-cli to prompt and attempt to update the app.json.
- This feature (like before) relies heavily on a git check to take place somewhere later in the command. AFAICT this always happens but it could change in the future, to remove this edge case we'd either need to aggressively check the git status after updating which I found to be pretty annoying, or move off of the git system for tracking code in the future.
- Added a better bundleIdentifier validation algorithm https://github.com/expo/expo-cli/pull/3355

# Test Plan

- Run `eas build` in a project without eas setup, and no `ios.bundleIdentifer` or `android.package` in the app.json, the user should be prompted to add the values. If the project has an app.config.js then the program will exit and the user will be informed what they need to add to their config file.
- Run `eas build` in a project without eas setup, and a mismatched `ios.bundleIdentifer` and `android.package` in the app.json, the user should be prompted to select which option to use and the native code or app.json should be sync'd.
